auth/services/contract/notary.go

```diff
@@ -116,23 +116,35 @@ func (n *notary) DrawUpContract(template contract.Template, orgID did.DID, valid
 		return nil, fmt.Errorf("could not draw up contract: organization is not managed by this node: %w", ErrMissingOrganizationKey)
 	}
 
-	// DrawUpContract draws up a contract for a specific organization from a template
-	result, err := n.vcr.Get(concept.OrganizationConcept, false, orgID.String())
-	if err != nil {
-		return nil, fmt.Errorf("could not draw up contract: %w", err)
+	searchTerms := []vcr.SearchTerm{
+		{IRIPath: jsonld.CredentialSubjectPath, Value: orgID.String()},
+		{IRIPath: jsonld.OrganizationNamePath, Type: vcr.NotNil},
+		{IRIPath: jsonld.OrganizationCityPath, Type: vcr.NotNil},
 	}
-	orgName, err := result.GetString(concept.OrganizationName)
+
+	result, err := n.vcr.Search(context.Background(), searchTerms, false, nil)
 	if err != nil {
-		return nil, fmt.Errorf("could not draw up contract, could not extract organization name: %w", err)
+		return nil, fmt.Errorf("could not find a credential: %w", err)
 	}
-	orgCity, err := result.GetString(concept.OrganizationCity)
+
+	if len(result) == 0 {
+		return nil, errors.New("could not find a trusted credential with an organization name and city")
+	}
+
+	// expand
+	reader := jsonld.Reader{DocumentLoader: n.contextManager.DocumentLoader()}
+	document, err := reader.Read(result[0])
```

Comment:
It could (at least for now). JSON-LD doesn't define if a field is of a certain type. For that we need to add a JSON schema. What we do now at this point of the code that at least one name and city exist.

Reply:
If there's multiple, the signed contract validation might break. Correct way would be to check against all organization VCs instead of the first one (I think @jelmerterwal encountered this). However, since this bug already exists, I'll create an issue.